File: python/Programmers/숫자 문자열과 영단어.py
def solution(s):
    answer = ''
    
    alph = {'zero': '0', 'one': '1', 'two': '2', 'three': '3', 'four': '4', 'five': '5', 'six': '6', 'seven': '7', 'eight': '8', 'nine': '9'}
    
    i = 0
    while i < len(s):
        # 숫자면 답에 더해줌
        if '0' <= s[i] <= '9':
            answer += s[i]
            i += 1
        else:
            # 문자면 문자를 식별하고 숫자로 바꿔서 답에 더해줌
            tmp = ''
            while i < len(s) and 'a' <= s[i] <= 'z':
                tmp += s[i]
                if alph.get(tmp):
                    answer += alph[tmp]
                    tmp = ''
                i += 1
            
    return int(answer)

File: Javascript/Programmers/숫자 문자열과 영단어.js
/*
영어로 된 숫자를 수로 변환해 원래 수를 반환하는 문제
*/

function solution(s) {
  var answer = '';
  
  const numbers = {
      'one': '1',
      'two': '2',
      'three': '3',
      'four': '4',
      'five': '5',
      'six': '6',
      'seven': '7',
      'eight': '8',
      'nine': '9',
      'zero': '0'
  }
  
  let word = ''
  
  for (i = 0; i < s.length; i += 1) {
      if (!isNaN(s[i])) {
          if (word in numbers) {
              answer += numbers[word]
              word = ''
          }
          answer += s[i]
      } else {
          
          if (word in numbers) {
              answer += numbers[word]
              word = ''
          }
          
          word += s[i]
      }
  }
  
  if (word) {
      answer += numbers[word]
  }
  
  return parseInt(answer);
}